DataFormats/PatCandidates/src/PackedCandidate.cc

```diff
@@ -150,7 +150,8 @@ float pat::PackedCandidate::dxy(const Point &p) const {
 float pat::PackedCandidate::dz(const Point &p) const {
     maybeUnpackBoth();
     const float phi = float(p4_.load()->Phi())+dphi_;
-    return (vertex_.load()->Z()-p.Z())  - ((vertex_.load()->X()-p.X()) * std::cos(phi) + (vertex_.load()->Y()-p.Y()) * std::sin(phi)) * p4_.load()->Pz()/p4_.load()->Pt();
+    const float pzpt = deta_ ? std::sinh(etaAtVtx()) : p4_.load()->Pz()/p4_.load()->Pt();
+    return (vertex_.load()->Z()-p.Z())  - ((vertex_.load()->X()-p.X()) * std::cos(phi) + (vertex_.load()->Y()-p.Y()) * std::sin(phi)) * pzpt;
```

Comment:
changes in this file are merged already as a part of #20491
it looks like in this PR 60f6f31 comes before 852157f while in #20491 the order is reversed and there is also 5db31d6fb in between, which is not also in this PR.
Some cleanup of the history would be nice.